app/src/main/java/org/oppia/android/app/home/WelcomeViewModel.kt

```diff
@@ -18,14 +19,14 @@ class WelcomeViewModel(
     oppiaClock
   ).getGreetingMessage()
 
+  // Overriding equals is needed so that DataProvider combine functions used in the HomeViewModel
+  // will only rebind when the actual data in the data list changes, rather than when the ViewModel
+  // object changes.
   override fun equals(other: Any?): Boolean {
-    if (this === other) {
-      return true
-    }
-    if (other == null || other.javaClass != javaClass) {
-      return false
-    }
-    val otherResult = other as WelcomeViewModel
-    return otherResult.profileName == this.profileName
+    return other is WelcomeViewModel &&
+      this.profileName == other.profileName &&
+      this.greeting == other.greeting
   }
+
+  override fun hashCode() = Objects.hash(this.profileName) + Objects.hash(this.greeting)
```

Comment:
``hash`` is variadic, so this should work: ``override fun hashCode() = Objects.hash(profileName, greeting)``

Reply:
Done (had to search what variadic meant... new vocab word!)